mlcp/src/main/java/com/marklogic/contentpump/LocalJobRunner.java

```diff
@@ -319,6 +317,32 @@ private int assignThreads(int splitIndex, int splitCount) {
             return threadCount / splitCount;
         }
     }
+
+    private void createThreadPool(Configuration conf) {
+        int numThreads;
+        if (threadCount != 0) {
+            // Use specified threadCount in the command line
+            numThreads = threadCount;
+        } else {
+            // Use server max thread count
+            numThreads =  conf.getInt(
+                MarkLogicConstants.SERVER_THREAD_COUNT, DEFAULT_THREAD_COUNT);
+            if (numThreads == 0) {
+                // ML server version is below 10.0-4.2, unable to get server
+                // thread count
+                numThreads = DEFAULT_THREAD_COUNT;
+            } else {
+                numThreads = (int)(0.5 * numThreads);
```

Comment:
Is this the heuristic for aligning mlcp (Java) threads with MarkLogic app server threads?

Reply:
Hi Justin,

Yes, this value is suggested by Silvano. We need further performance testing to decide on the best heuristic to use.

Thanks,
Vanessa